tests/test_lightningd.py

```diff
@@ -288,7 +288,7 @@ class LightningDTests(BaseLightningDTests):
     def connect(self, may_reconnect=False):
         l1 = self.node_factory.get_node(may_reconnect=may_reconnect)
         l2 = self.node_factory.get_node(may_reconnect=may_reconnect)
-        ret = l1.rpc.connect(l2.info['id'], 'localhost', l2.info['port'])
+        ret = l1.rpc.connect(l2.info['id'], 'localhost', l2.info['address'][0]['port'])
```

Comment:
We should just shortcut this and make the `port` an attribute of the `LightningNode`, no need to keep the structure of the `getinfo` call

Reply:
We should actually use an ephemeral port for testing.

I thing we should enhance --addr to allow:
1. Hostname variants: 
    * `` (empty string) - All IPv4 & IPv6 interfaces.  This would be default unless of course --autolisten=0.  Announce if addresses seem routable.
    * `0.0.0.0`: All IPv4 interfaces.  Announce if address seems routable.
    * `::`: All IPv6 interfaces.  Announce if address seems routable.
2. Port variants:
    * ``: (empty string) default port, ie. 9735.
    * `:0`: an ephemeral port.

This is more of a rework though, since guess_addresses needs to move into gossipd, when then needs to feed the actual address results *back* to the master for getinfo.